src/main/java/org/zephyrsoft/sdb2/gui/MainWindow.java

```diff
@@ -1485,16 +1545,6 @@ public void mouseClicked(MouseEvent e) {
 		panelEdit.add(scrollPaneLyrics, gbcScrollPaneLyrics);
 		
 		editorLyrics = new JEditorPane();
-		editorLyrics.addFocusListener(new FocusAdapter() {
-			@Override
-			public void focusLost(FocusEvent e) {
-				try {
-					handleSongDataFocusLost();
-				} catch (Throwable ex) {
-					handleError(ex);
-				}
-			}
-		});
```

Comment:
Why are these focus listeners not necessary anymore?

Reply:
My implemented approach:

* The opened Song is a copy of the one inside the model.
* Changes are handled by the SongsModelController
* A song is saved and the listview is updated on Ctrl+S and WindowFocus, TabFocus changes.

Instead of creating an empty song in the SongsModel and reindexing and sorting the wohle listview on every editfield focus change. (And also sending a patch to all connected instances and the db)

Reply:
I'm afraid that the typical user will not know that he/she has to press Ctrl+S to update the list, there will only be wondering why the title is not displayed in the list although it was typed in (and is visible clearly right of the list). This confusion will probably result in doubt in the software.

If the problem only was that too many patches were sent, please re-add the focus listeners so the UI is consistent for the user. I don't think the problem was that the UI was slow as the reindexing is always done in a background executor.

Plus the model should always contain the most up-to-date data because the SDB sometimes is not closed before shutting down the computer - in this case, it only has very little time to save the database and it would be beneficial to already have the current data.

Reply:
@l-spiecker Could you please (re-)implement means to save the song data even if no tab or window focus change happens (maybe additionally to window and tab focus listeners)?

As I wrote, the users typically don't press Ctrl-S after changing something, they might just stay there and do nothing - maybe until the computer is shut down (which would be no problem when the Windows start menu is used as this should fire a window change event, but it's also possible to use the hardware button). It could even be a timed periodic check if the song has to be saved.

This is not to annoy you, I'm just afraif of data loss. This was a big topic in the beginning of the project - a burnt child dreads the fire...

Reply:
Yes, I understand that. I already started implementing that after your last message. I will try to push the changes this week.